linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -658,13 +658,16 @@ final class Analyzer(config: CommonPhaseConfig, initial: Boolean,
     private[this] val _instantiatedFrom = new AtomicReference[List[From]](Nil)
     def instantiatedFrom: List[From] = _instantiatedFrom.get()
 
-    val dispatchCalledFrom: mutable.Map[MethodName, List[From]] = mutable.Map.empty
+    private[this] val _dispatchCalledFrom: mutable.Map[MethodName, AtomicReference[List[From]]] = emptyThreadSafeMap
+    def dispatchCalledFrom(methodName: MethodName): Option[List[From]] =
+      _dispatchCalledFrom.get(methodName).map(_.get())
 
     /** List of all instantiated (Scala) subclasses of this Scala class/trait.
      *  For JS types, this always remains empty.
      */
-    var instantiatedSubclasses: List[ClassInfo] = Nil
-    var methodsCalledLog: List[MethodName] = Nil
+    private val _instantiatedSubclasses = new AtomicReference[List[ClassInfo]](Nil)
+
+    private val methodsCalledLog = new AtomicReference[List[MethodName]](Nil)
```

Comment:
Note: This is always strictly `methodsCalledLog.toSet == dispatchCalledFrom.keySet`. However, we need to be able to snapshot this in `subclassInstantiated`. `TrieMap` supports snapshotting, but `mutable.Map` doesn't.

So the options are:
- Keep this.
- Switch to `AtomicReference[immutable.Map[MethodName, AtomicReference[List[From]]]` (I have concerns about key contention on the map here).
- Do some type hackery so we can get "snapshots" also on normal mutable maps in JS.